File: exercises/day13-Keyboard-First-Refactoring/java/src/main/java/sleightrip/elf/ElfSession.java
package sleightrip.elf;

import sleightrip.exception.CollaboratorCallException;

public class ElfSession {

    private static ElfSession instance = new ElfSession();

    public static ElfSession getInstance() {
        return instance;
    }

    public Elf getLoggedUser() {
        throw new CollaboratorCallException(
                "ElfSession is a static collaborator and should not be used directly in unit tests"
        );
    }
}

File: exercises/day13-Keyboard-First-Refactoring/scala/src/main/scala/sleightrip/elf/ElfSession.scala
package sleightrip.elf

import sleightrip.exception.CollaboratorCallException

object ElfSession {
  private val instance = new ElfSession

  def getInstance(): ElfSession = instance
}

class ElfSession private {
  def getLoggedUser: Elf = {
    throw new CollaboratorCallException(
      "ElfSession is a static collaborator and should not be used directly in unit tests"
    )
  }
}
